IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -235,129 +216,148 @@ - (void)executeRequestImpl:(MSIDRequestCompletionBlock)completionBlock
             }
         }
     }
+    
+    [self fetchCachedRefreshTokenWithCompletionHandler:^(MSIDBaseToken<MSIDRefreshableToken> *refreshToken, MSIDRefreshTokenTypes tokenType, NSError *error) {
+        if (error)
+        {
+            completionBlock(nil, error);
+            return;
+        }
+        
+        [self tryRefreshToken:refreshToken tokenType:tokenType completionBlock:completionBlock];
+    }];
+    
+}
 
-    NSError *frtCacheError = nil;
-    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Looking for Family Refresh token...");
-    MSIDRefreshToken *familyRefreshToken = [self familyRefreshTokenWithError:&frtCacheError];
-
-    if (frtCacheError)
+- (void)fetchCachedRefreshTokenWithCompletionHandler:(void (^)(MSIDBaseToken<MSIDRefreshableToken> *, MSIDRefreshTokenTypes, NSError *))completionHandler
```

Comment:
nit: This seems to be a synchronous operation. Does it actually need a completion block? You can return refresh token and pass NSError as a parameter.

Reply:
Good point, the reason I used a completion block here is due to another params called  ```MSIDRefreshTokenTypes```, where would need to distinguish the type of refresh in later code path